node/silkworm/stagedsync/stagedsync.hpp

```diff
@@ -22,31 +22,94 @@
 #include <filesystem>
 #include <vector>
 
+#include <silkworm/consensus/engine.hpp>
+#include <silkworm/db/stages.hpp>
 #include <silkworm/db/tables.hpp>
-#include <silkworm/stagedsync/util.hpp>
+#include <silkworm/execution/analysis_cache.hpp>
+#include <silkworm/execution/state_pool.hpp>
+#include <silkworm/stagedsync/common.hpp>
+#include <silkworm/stagedsync/recovery/recovery_farm.hpp>
 
 namespace silkworm::stagedsync {
 
-inline constexpr size_t kDefaultBatchSize = 512_Mebi;
+class BlockHashes final : public IStage {
+  public:
+    explicit BlockHashes(NodeSettings* node_settings) : IStage(db::stages::kBlockHashesKey, node_settings){};
+    ~BlockHashes() override = default;
+
+    StageResult forward(db::RWTxn& txn) final;
+    StageResult unwind(db::RWTxn& txn, BlockNum to) final;
+    StageResult prune(db::RWTxn& txn) final;
+    std::vector<std::string> get_log_progress() final;
+
+  private:
+    std::unique_ptr<etl::Collector> collector_{nullptr};
+
+    /* Stats */
+    uint16_t current_phase_{0};
+    BlockNum reached_block_num_{0};
+};
+
+class Senders final : public IStage {
+  public:
+    explicit Senders(NodeSettings* node_settings) : IStage(db::stages::kSendersKey, node_settings){};
+    ~Senders() override = default;
+
+    StageResult forward(db::RWTxn& txn) final;
+    StageResult unwind(db::RWTxn& txn, BlockNum to) final;
+    StageResult prune(db::RWTxn& txn) final;
+    std::vector<std::string> get_log_progress() final;
+
+  private:
+    std::unique_ptr<recovery::RecoveryFarm> farm_{nullptr};
+};
+
+class Execution final : public IStage {
```

Comment:
I'd suggest to use names like `ExecutionStage`, `SendersStage`, etc for clarity.

Reply:
And I'd move this class declaration to `stage_execution.hpp` to follow the traditional pattern of `stage_execution.cpp` implementing methods from the corresponding header.